backup.sh

```diff
@@ -26,53 +27,75 @@
 #
 # Authors:
 #  Cyrill von Wattenwyl <cyrill.vonwattenwyl@adfinis.com>
-
+#  Valentin Maillot <valentin.maillot@adfinis.com>
 
 set -xeuo pipefail
 
-# set proper umask
-umask "${OCP_BACKUP_UMASK}"
+# check storage type
+if [ "${OCP_BACKUP_S3}" = "true" ]; then
+    # prepare & push backup to S3
 
-# validate expire type
-case "${OCP_BACKUP_EXPIRE_TYPE}" in
-    days|count|never) ;;
-    *) echo "backup.expiretype needs to be one of: days,count,never"; exit 1 ;;
-esac
+    # update CA trust
+    update-ca-trust
 
-# validate  expire numbers
-if [ "${OCP_BACKUP_EXPIRE_TYPE}" = "days" ]; then
-  case "${OCP_BACKUP_KEEP_DAYS}" in
-    ''|*[!0-9]*) echo "backup.expiredays needs to be a valid number"; exit 1 ;;
-    *) ;;
-  esac
-elif [ "${OCP_BACKUP_EXPIRE_TYPE}" = "count" ]; then
-  case "${OCP_BACKUP_KEEP_COUNT}" in
-    ''|*[!0-9]*) echo "backup.expirecount needs to be a valid number"; exit 1 ;;
-    *) ;;
-  esac
-fi
+    # configure mcli assuming the bucket already exists
+    bash +o history
+    mcli alias set "${OCP_BACKUP_S3_NAME}" "${OCP_BACKUP_S3_HOST}" "${OCP_BACKUP_S3_ACCESS_KEY}" "${OCP_BACKUP_S3_SECRET_KEY}"
+    bash -o history
+
+    # create backup to temporary location
+    chroot /host /usr/local/bin/cluster-backup.sh /var/tmp/etcd-backup
+
+    # move files to S3 and delete temporary files
+    mcli mv /host/var/tmp/etcd-backup/* "${OCP_BACKUP_S3_NAME}"/"${OCP_BACKUP_S3_BUCKET}"
+    rm -rv /host/var/tmp/etcd-backup
```

Comment:
Where/how are old backups removed from the object storage bucket?

Reply:
OK, found it in the README. It would be helpful if the steps to enable retention on the bucket could be added  to the README (e.g. with an example for MinIO).

Reply:
done! thanks for the tips